test/test-spec-tests.js

```diff
@@ -0,0 +1,91 @@
+/* -*- Mode: js; js-indent-level: 2; -*- */
+/*
+ * Copyright 2024 Mozilla Foundation and contributors
+ * Licensed under the New BSD license. See LICENSE or:
+ * http://opensource.org/licenses/BSD-3-Clause
+ */
+
+const fs = require('node:fs/promises');
+const SourceMapConsumer =
+  require("../lib/source-map-consumer").SourceMapConsumer;
+
+const sourceMapSpecTests = require("./source-map-tests/source-map-spec-tests.json");
+
+async function readJSON(path) {
+  const file = await fs.open(require.resolve(path));
+  const json = JSON.parse(await file.readFile());
+  file.close();
+  return json;
+}
+
+// Known failures due to intentional implementation choices or due to bugs.
+const skippedTests = [
+  // Versions are explicitly checked a bit loosely.
+  "versionNumericString",
+  // Stricter sources array checking isn't implemented.
+  "sourcesNotStringOrNull",
+  "sourcesAndSourcesContentBothNull",
+  // Stricter names array checking isn't implemented.
+  "namesMissing",
+  "namesNotString",
+  // This check isn't as strict in this library.
+  "invalidMappingNotAString1",
+  // A mapping segment with no fields is technically invalid in the spec.
+  "invalidMappingSegmentWithZeroFields",
+  // These tests fail due to imprecision in the spec about the 32-bit limit.
+  "invalidMappingSegmentWithColumnExceeding32Bits",
+  "invalidMappingSegmentWithOriginalLineExceeding32Bits",
+  "invalidMappingSegmentWithOriginalColumnExceeding32Bits",
+  // A large VLQ that should parse, but currently does not.
+  "validMappingLargeVLQ",
+  // The library currently doesn't check the types of offset lines/columns.
+  "indexMapOffsetLineWrongType",
+  "indexMapOffsetColumnWrongType",
+  // The spec is not totally clear about this case.
+  "indexMapInvalidBaseMappings",
+  // The spec's definition of overlap can be refined
+  "indexMapInvalidOverlap",
+  // Not clear if this test makes sense, but spec isn't clear on behavior
+  "validMappingNullSources"
+]
```

Comment:
Are these skipped tests specific to the mozilla sourcemap implementation
```suggestion
// Known failures due to intentional implementation choices or due to bugs.
const skippedTests = [
  // Versions are explicitly checked a bit loosely.
  "versionNumericString",
  // Stricter sources array checking isn't implemented.
  "sourcesNotStringOrNull",
  "sourcesAndSourcesContentBothNull",
  // Stricter names array checking isn't implemented.
  "namesMissing",
  "namesNotString",
  // This check isn't as strict in this library.
  "invalidMappingNotAString1",
  // A mapping segment with no fields is technically invalid in the spec.
  "invalidMappingSegmentWithZeroFields",
  // These tests fail due to imprecision in the spec about the 32-bit limit.
  "invalidMappingSegmentWithColumnExceeding32Bits",
  "invalidMappingSegmentWithOriginalLineExceeding32Bits",
  "invalidMappingSegmentWithOriginalColumnExceeding32Bits",
  // A large VLQ that should parse, but currently does not.
  "validMappingLargeVLQ",
  // The library currently doesn't check the types of offset lines/columns.
  "indexMapOffsetLineWrongType",
  "indexMapOffsetColumnWrongType",
  // The spec is not totally clear about this case.
  "indexMapInvalidBaseMappings",
  // The spec's definition of overlap can be refined
  "indexMapInvalidOverlap",
  // Not clear if this test makes sense, but spec isn't clear on behavior
  "validMappingNullSources"
]
```

Reply:
Yes, for example the "versionNumericString" test is skipped because this library specifically accepts the case where the version field is a number in a string. Other implementations have different behavior (e.g., the [source map validator](https://github.com/jkup/source-map-validator) uses this library but layers more strict checking on top).